source/src_lcao/ELEC_scf.cpp

```diff
@@ -500,6 +517,216 @@ void ELEC_scf::scf(const int& istep,
 				*/
 			}
 
+			if (conv_elec && ELEC_evolve::tddft && istep >= 2)
+        	{
+			#ifdef __MPI
+			//#else
+			///*
+				for(int ik=0; ik<GlobalC::kv.nks; ik++)
+				{
+					/*
+					cout<<"nbands="<<GlobalV::NBANDS<<" nlocal="<<GlobalV::NLOCAL<<endl;
+					cout<<"ncol="<<uhm.LM->ParaV->ncol<<"ncol_bands="<<uhm.LM->ParaV->ncol_bands<<" nrow="<<uhm.LM->ParaV->nrow<<endl;;
+					cout<<"nloc_wfc="<<uhm.LM->ParaV->nloc_wfc<<" nloc="<<uhm.LM->ParaV->nloc<<endl;
+					*/
+					complex<double>* tmp1 = new complex<double> [uhm.LM->ParaV->nloc];
+					complex<double>* tmp2 = new complex<double> [uhm.LM->ParaV->nloc];
+					complex<double>* tmp3 = new complex<double> [uhm.LM->ParaV->nloc_wfc];
+					complex<double>* Htmp = new complex<double> [uhm.LM->ParaV->nloc];
```

Comment:
Allocating memory each time in for-loop might be time consuming. Consider move temp vars outside the loop?